.github/workflows/ci.yml

```diff
@@ -0,0 +1,70 @@
+name: Continuous Integration
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  fmt:
+    name: Rustfmt
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+      - run: rustup component add rustfmt
+      - uses: actions-rs/cargo@v1
+        with:
+          command: fmt
+          args: --all -- --check
+  clippy:
+    name: Clippy
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+      - run: rustup component add clippy
+      - uses: actions-rs/cargo@v1
+        with:
+          command: clippy
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
```

Comment:
where is this matrix being used?

Reply:
So matrix is used to run build twice using different versions of rust. One using the stable version and other for the minimum supported version.

Reply:
But the matrix is not being referred anywhere...?

Reply:
Sorry, my bad. There were couple of errors in the CI workflow. I have fixed them and updated a PR.